src/formatDate.js

```diff
@@ -48,9 +48,51 @@
  *
  * @returns {string}
  */
+const CENTURY_INDICATOR = 30;
+const CENTURY_21 = '20';
+const CENTURY_20 = '19';
+const YEAR_FORMAT_OF_4 = 'YYYY';
+const YEAR_FORMAT_OF_2 = 'YY';
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateArray = date.split(fromSeparator);
+  const newDateArray = [];
+
+  formatYear(fromFormat, toFormat, dateArray);
+
+  for (let i = 0; i < dateArray.length; i++) {
+    newDateArray[i] = dateArray[fromFormat.indexOf(toFormat[i])];
+  }
+
+  return newDateArray.join(toSeparator);
+}
+
+function formatYear(fromFormat, toFormat, dateArray) {
+  const fromYearIndex = fromFormat.indexOf(YEAR_FORMAT_OF_2)
+  + fromFormat.indexOf(YEAR_FORMAT_OF_4) + 1;
+  const toYearIndex = toFormat.indexOf(YEAR_FORMAT_OF_2)
+  + toFormat.indexOf(YEAR_FORMAT_OF_4) + 1;
+
+  if (fromFormat[fromYearIndex] === toFormat[toYearIndex]) {
+    return;
+  }
+
+  fromFormat[fromYearIndex] = toFormat[toYearIndex];
+
+  if (fromFormat[fromYearIndex] === YEAR_FORMAT_OF_4) {
+    if (+dateArray[fromYearIndex] < CENTURY_INDICATOR) {
+      dateArray[fromYearIndex] = CENTURY_21 + dateArray[fromYearIndex];
```

Comment:
`dateArray[fromYearIndex]` is repeated many times, this can be put into a variable and reused.

Reply:
through this expression, I change the value inside of the array, if I create a variable, both the variable and the direct access to the array element through this entry will appear in my code. I think it will confuse more then now.